$NetBSD: patch-CMakeLists.txt,v 1.3 2022/03/13 15:22:32 tnn Exp $

Put headers in subdirectory to avoid conflict with clang.
Don't implement sys/regset.h workaround, fix source instead.
Don't build locstats.

--- CMakeLists.txt.orig	2021-06-28 16:23:38.000000000 +0000
+++ CMakeLists.txt
@@ -902,7 +902,7 @@ endif()
 
 if( ${CMAKE_SYSTEM_NAME} MATCHES SunOS )
    # special hack for Solaris to handle crazy system sys/regset.h
-   include_directories("${LLVM_MAIN_INCLUDE_DIR}/llvm/Support/Solaris")
+   #include_directories("${LLVM_MAIN_INCLUDE_DIR}/llvm/Support/Solaris")
 endif( ${CMAKE_SYSTEM_NAME} MATCHES SunOS )
 
 # Make sure we don't get -rdynamic in every binary. For those that need it,
@@ -1027,7 +1027,7 @@ endif()
 
 if (NOT LLVM_INSTALL_TOOLCHAIN_ONLY)
   install(DIRECTORY include/llvm include/llvm-c
-    DESTINATION include
+    DESTINATION include/libLLVM
     COMPONENT llvm-headers
     FILES_MATCHING
     PATTERN "*.def"
@@ -1038,7 +1038,7 @@ if (NOT LLVM_INSTALL_TOOLCHAIN_ONLY)
     )
 
   install(DIRECTORY ${LLVM_INCLUDE_DIR}/llvm ${LLVM_INCLUDE_DIR}/llvm-c
-    DESTINATION include
+    DESTINATION include/libLLVM
     COMPONENT llvm-headers
     FILES_MATCHING
     PATTERN "*.def"
@@ -1125,5 +1125,5 @@ if (LLVM_INCLUDE_BENCHMARKS)
 endif()
 
 if (LLVM_INCLUDE_UTILS AND LLVM_INCLUDE_TOOLS)
-  add_subdirectory(utils/llvm-locstats)
+#  add_subdirectory(utils/llvm-locstats)
 endif()
